### PR TITLE
For #88: Location sync logic

### PR DESCRIPTION
**Pull request by paulodamaso**
For #88: Implemented `Location` sync logic

* Implemented `sync.py` `sync_location` logic to sync the objects received
* Corrected test to compare objects correctly
* Left puzzle for database implementation

### CODE REVIEW

**Comment by paulodamaso** [comment]
@0crat status

**Comment by 0crat** [comment]
Job [#146](https://github.com/timelesslounge/timelessis/issues/146) is now in scope, role is `REV`

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/146#issuecomment-461380834))

@paulodamaso This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#146](https://github.com/timelesslounge/timelessis/issues/146) is [in scope](http://www.zerocracy.com/policy.html#14) for 14min
  * The role is `REV`
  * The job is not assigned to anyone
  * The [budget](http://www.zerocracy.com/policy.html#4) is not set yet
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23146%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by 0crat** [comment]
This pull request [#146](https://github.com/timelesslounge/timelessis/issues/146) is assigned to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/d6893614-e28f-4350-93f7-6c2bc7a0d5b3); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/test_sync.py [hunk 1]
```diff
@@ -76,7 +78,25 @@ def setup_class(cls):
             url="http://localhost:{port}".format(port=cls.port)
         )
 
-    @unittest.skip("sync.sync_location not implemented yet")
+
     def test_sync_location(self):
         self.poster_sync.sync_location(self.poster, self.company)
-        assert (self.poster.locations() == self.company.locations)
+        for location in self.company.locations:
+            for loc in self.poster.locations():
```
Comment (vladarefiev) [inline tests/test_sync.py]: @paulodamaso confusing, let's call it `poster_location` for instance

File: timeless/sync/sync.py [hunk 2]
```diff
@@ -1,13 +1,35 @@
...
+     start by creating structure for integration tests with database
 
 """
 class PosterSync():
 
     def sync_location(poster, company):
-        raise Exception("sync_location not implemented yet")
+        synced =[]
+        for location in company.locations:
+            for loc in poster.locations():
```
Comment (vladarefiev) [inline timeless/sync/sync.py]: @paulodamaso the same

File: timeless/sync/sync.py [hunk 3]
```diff
@@ -1,13 +1,35 @@
...
+                            city=loc["city"],
+                            address=loc["address"],
+                            longitude=loc["longitude"],
+                            latitude=loc["latitude"],
+                            type=loc["type"],
+                            status=loc["status"],
+                            comment=loc["comment"]
+                        )
+                    )
+        company.locations = synced
```
Comment (vladarefiev) [inline timeless/sync/sync.py]: @paulodamaso Why you didn't save it to database?

Comment (paulodamaso) [reply timeless/sync/sync.py]: @vladarefiev I've kept the sync logic here, decoupled from data storage. The tests in `test_sync.py` where made decoupled from database, and to test them against it I would have to do a lot of mocks and / or create all the necessary data demanded by `Location` table, and create an integration testing logic for the project. This way we test if the sync routine is syncing the object, and we can make only the integration test later.

Comment (vladarefiev) [reply timeless/sync/sync.py]: @paulodamaso got it. thanks

**Comment by vladarefiev** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 15 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vladarefiev[/z](https://www.0crat.com/u/vladarefiev), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#146](https://github.com/timelesslounge/timelessis/issues/146) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@vladarefiev According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two cosmetic issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vladarefiev** [comment]
@ypshenychka confirm

**Comment by ypshenychka** [comment]
@vladarefiev thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +20 point(s) just awarded to @vladarefiev[/z](https://www.0crat.com/u/vladarefiev)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**